src/lib.rs

```diff
@@ -210,21 +219,31 @@ impl Distribution<XorName> for Standard {
 #[allow(clippy::large_enum_variant)]
 #[derive(Hash, Eq, PartialEq, PartialOrd, Ord, Clone, Serialize, Deserialize)]
 pub enum Message {
+    /// Request with the message ID.
     Request {
+        /// Request.
         request: Request,
+        /// Associated message ID.
         message_id: MessageId,
+        /// Optional signature.
```

Comment:
Why is it optional? Should be expanded into something like
```suggestion
        /// Signature of `(request, message_id)`. Optional if the request is read-only.
```